smooth/framework/functions/functions.py

```diff
@@ -42,3 +42,119 @@ def get_sim_time_span(n_interval, step_size):
+                nb_intervals = len(component_flows[flow])
+                nb_trailing_none = nb_intervals
+                for flow_val in component_flows[flow]:
+                    if flow_val is not None:
+                        nb_trailing_none -= 1
+                    else:
+                        break
+                # check if it's a chp component which consists of two oemof models
+                # if so get rid of the ending '_electric' or '_thermal'
+                if this_flow_name_split[0][-9:] == '_electric':
+                    this_flow_name_split[0] = this_flow_name_split[0][:-9]
+                elif this_flow_name_split[0][-8:] == '_thermal':
+                    this_flow_name_split[0] = this_flow_name_split[0][:-8]
+                if this_flow_name_split[0] == component_result.name:
+                    # Case: Component flows into bus.
+                    bus = this_flow_name_split[1]
+                    # Check if this component already has a flow with this bus.
+                    if bus in this_comp_flows:
+                        updated_bus_list = []
+                        for i_val in range(len(this_comp_flows[bus]) - nb_trailing_none):
+                            # Get the summed up value.
+                            this_val = this_comp_flows[bus][i_val] + component_flows[flow][i_val]
+                            updated_bus_list.append(this_val)
+                        # Override the old bus list with the updated one.
+                        this_comp_flows[bus] = updated_bus_list
+                    else:
+                        # Case: Component has no flow with this bus yet.
+                        this_comp_flows[bus] = component_flows[flow][:nb_intervals - nb_trailing_none]
+
+                else:
+                    # Case: Component takes from bus.
+                    bus = this_flow_name_split[0]
+                    # Check if this component already has a flow with this bus.
+                    if bus in this_comp_flows:
+                        updated_bus_list = []
+                        for i_val in range(len(this_comp_flows[bus]) - nb_trailing_none):
+                            # Get the summed up value.
+                            this_val = this_comp_flows[bus][i_val] - component_flows[flow][i_val]
+                            updated_bus_list.append(this_val)
+                        # Override the old bus list with the updated one.
+                        this_comp_flows[bus] = updated_bus_list
+                    else:
+                        # Case: Component has no flow with this bus yet.
+                        this_comp_flows[bus] = [-this_val for this_val in
+                                                component_flows[flow][:nb_intervals - nb_trailing_none]]
+
+            for this_bus in this_comp_flows:
+                # Replaces shorthand component names in the results with the official names for those listed.
```

Comment:
Might be done in dictionary. Translating strings is the literal definition of a dictionary.

Reply:
Already have an issue #79 for that
It still does not make sense to have a dictionary hard coded, as the user can freely choose the names. So it has to be done somewhere else anyways

Reply:
Came up with an optional argument label dict for the plotting function, this way it is not hard coded anymore
https://github.com/rl-institut/smooth/pull/77/commits/5394241d8718615e498fb74331dab14a193ad4a3